pysqream_blue/cursor.py

```diff
@@ -62,17 +73,15 @@ def execute(self, query: str, params=None):
     def cancel(self):
         return self._request_cancel()
 
+    def _request_cancel_inner(self):
+        cancel_response: qh_messages.CancelResponse = self._cancel()
+
     def _request_cancel(self):
 
         if self.stmt_id is None:
             self.logs.log_and_raise(ProgrammingError, "Context Id is not found")
 
-        cancel_response = None
-        try:
-            cancel_response: qh_messages.CancelResponse = self._cancel()
-        except grpc.RpcError as rpc_error:
-            self.logs.log_and_raise(ProgrammingError,
-                          f'Context id: {self.stmt_id}. Error from grpc while attempting to cancel the query.\n{rpc_error}')
+        cancel_response = self._retry_function_on_stream_error(self._request_cancel_inner, "cancel the query")
```

Comment:
you can send only "cancel", and on the error log add "the query" after the function action.